### ISSUE TITLE
LibWeb: CSSParser: assertion failed on consume_specific / consume_one

### ISSUE THREAD

**Issue by bcoles** [SerenityOS/serenity#4232]
`Browser` crashes when trying to `peek()` into malformed CSS. Malformed CSS should not be a fatal error.

Given the wildly inconsistent state of modern HTML/JS/CSS, compounded by competing standards and traditionally lax parsing by modern browsers, it is likely that malformed CSS will be encountered in the wild.

Reproduce:

```html
<style>{
```

Log:

```
Browser(24): FrameLoader::load: http://172.16.191.165/css.html
ProtocolServer(25): TCPSocket{0x028f0a18} connection appears to have closed in receive().
ProtocolServer(25): TCPSocket{0x028f0718} connection appears to have closed in receive().
Browser(24): I believe this content has MIME type 'text/html', encoding 'utf-8'
Browser(24): Parse error! void Web::HTML::HTMLDocumentParser::handle_initial(Web::HTML::HTMLToken&) @ 356
Browser(24): Parse error! void Web::HTML::HTMLDocumentParser::handle_text(Web::HTML::HTMLToken&) @ 1884
Browser(24): peek() != ':'
Browser(24): CSS parse error
Browser(24): CSS PARSER ASSERTION FAILED: index < css.length()
Browser(24): At character# 2 in CSS: _{
_
USERSPACE(24) ASSERTION FAILED: false
../Libraries/LibWeb/CSS/Parser/CSSParser.cpp:441
[Browser(24:24)]: Terminating Browser(24) due to signal 6
[FinalizerTask(3:3)]: CPU[0] NP(error) fault at invalid address V0x00f025ec
[FinalizerTask(3:3)]: copy_from_user(0xc094a92c, 0x00f025ec, 4) failed at V0x00f025ec
[FinalizerTask(3:3)]: 0xdeadc0de  Kernel::Processor::switch_context(Kernel::Thread*&, Kernel::Thread*&) +181
0xdeadc0de  Kernel::Scheduler::context_switch(Kernel::Thread*) +181
0xdeadc0de  Kernel::Scheduler::pick_next() +2280
0xdeadc0de  Kernel::Processor::check_invoke_scheduler() +125
0xdeadc0de  Kernel::Thread::die_if_needed() +157
0xdeadc0de  syscall_handler +1055
0xdeadc0de  syscall_asm_entry +49
0x0824aad4  raise +22
0x080480bd  abort +13
0x082460ed  tolower +0
0x080f1590  Web::CSSParser::consume_specific(char) +336
0x080f2647  Web::CSSParser::parse_property() +227
0x080f280a  Web::CSSParser::parse_declaration() +24
0x080f2954  Web::CSSParser::parse_rule() +190
0x080f2acc  Web::CSSParser::parse_sheet() +32
0x080f0445  Web::parse_css(Web::CSS::ParsingContext const&, AK::StringView const&) +230
0x080850b5  Web::HTML::HTMLStyleElement::children_changed() +255
0x08088119  Web::HTML::HTMLDocumentParser::flush_character_insertions() +117
0x0809ab6d  Web::HTML::HTMLDocumentParser::run(AK::URL const&) +359
0x0809af16  Web::HTML::parse_html_document(AK::StringView const&, AK::URL const&, AK::String const&) +64
0x080d1ca0  Web::FrameLoader::create_document_from_mime_type(AK::ByteBuffer const&, AK::URL const&, AK::String const&, AK::String const&) +72
0x080d3b4d  Web::FrameLoader::resource_did_load() +399
0x080d5e9d  Web::Resource::for_each_client(AK::Function<void (Web::ResourceClient&)>) +309
0x080d6487  Web::Resource::did_load(AK::Badge<Web::ResourceLoader>, AK::ByteBuffer const&, AK::HashMap<AK::String, AK::String, AK::CaseInsensitiveStringTraits> const&) +1191
0x080d6f98  AK::Function<void (AK::ByteBuffer const&, AK::HashMap<AK::String, AK::String, AK::CaseInsensitiveStringTraits> const&)>::CallableWrapper<Web::ResourceLoader::load_resource(Web::Resource::Type, Web::LoadRequest const&)::{lambda(auto:1&, auto:2&)#1}>::call(AK::ByteBuffer const&, AK::HashMap<AK::String, AK::String, AK::CaseInsensitiveStringTraits> const&) const +46
0x080d77d1  AK::Function<void (bool, AK::ByteBuffer const&, AK::RefPtr<AK::SharedBuffer, AK::RefPtrTraits<AK::SharedBuffer> >, AK::HashMap<AK::String, AK::String, AK::CaseInsensitiveStringTraits> const&, AK::Optional<unsigned int>)>::CallableWrapper<Web::ResourceLoader::load(Web::LoadRequest const&, AK::Function<void (AK::ByteBuffer const&, AK::HashMap<AK::String, AK::String, AK::CaseInsensitiveStringTraits> const&)>, AK::Function<void (AK::String const&)>)::{lambda(bool, AK::ByteBuffer const&, auto:1, auto:2&, auto:3)#4}>::call(bool, AK::ByteBuffer const&, AK::RefPtr<AK::SharedBuffer, AK::RefPtrTraits<AK::SharedBuffer> >, AK::HashMap<AK::String, AK::String, AK::CaseInsensitiveStringTraits> const&, AK::Optional<unsigned int>) const +215
0x0812fc1c  Protocol::Download::did_finish(AK::Badge<Protocol::Client>, bool, AK::Optional<unsigned int>, unsigned int, int, IPC::Dictionary const&) +742
0x0812c324  Protocol::Client::handle(Messages::ProtocolClient::DownloadFinished const&) +254
0x0812cd81  ProtocolClientEndpoint::handle(IPC::Message const&) +53
0x0812f73f  IPC::Connection<ProtocolClientEndpoint, ProtocolServerEndpoint>::handle_messages() +123
0x0823c0a5  Core::EventLoop::pump(Core::EventLoop::WaitMode) +307
0x0823c375  Core::EventLoop::exec() +43
0x08048cda  main +2266
0x080509ee  _start +94

[FinalizerTask(3:3)]: Scheduler[0]: Reaped unparented process Browser(24), exit status: 6
```

**Comment by awesomekling** [comment]
There's an effort to write a spec compliant CSS parser in #2628.
While it's good to prevent crashes in the current ad-hoc parser, it will eventually be replaced. :)

**Comment by linusg** [comment]
Fixed in ec89302fbe42987f037eb725d34a59a4936c5925.
